airavata-services/profile-service/profile-service-server/src/main/java/org/apache/airavata/service/profile/handlers/TenantProfileServiceHandler.java

```diff
@@ -72,7 +72,9 @@ public String getAPIVersion(AuthzToken authzToken) throws TenantProfileServiceEx
     @SecurityCheck
     public String addGateway(AuthzToken authzToken, Gateway gateway) throws TenantProfileServiceException, AuthorizationException, TException {
         try {
-            gateway = tenantProfileRepository.create(gateway);
+            if (!checkDuplicate(gateway)) {
```

Comment:
If there is a duplicate then I think the code should throw a TenantProfileServiceException with an error message explaining that there is a duplicate.